src/functionaltests/resources/features/restEndpoints.feature

```diff
@@ -0,0 +1,69 @@
+    Then Request should get response code <responsecode>
+
+  # Note: 404 responses are when no data in database was found,
+  # since this scenario is not about fetching data. 
+  Examples: 
+    | responsecode | endpoint                               |
+    | 200          | /information                           |
+    | 200          | /auth                                  |
+    | 200          | /auth/login                            |
+    | 200          | /auth/checkStatus                      |
+    | 200          | /download                              |
+    | 200          | /download/eventsTemplate               |
+    | 200          | /download/rulesTemplate                |
+    | 200          | /download/subscriptionsTemplate        |
+    | 404          | /queryMissedNotifications/subs_name    |
+    | 404          | /queryAggregatedObject/id              |
+    | 200          | /rules                                 |
+    | 200          | /rules/rule-check/testRulePageEnabled  |
+
+  # This scenario and the Scenario Outline below does the same thing, keep one remove one before merge.
+  Scenario: Test POST PUT GET DELETE on /subscriptions endpoint
+    Given A POST request is prepared
+    And Subscription with name "test_subscription" is added as body
+    When Perform request on endpoint "/subscriptions"
+    Then Request should get response code 200
+    Given A PUT request is prepared
+    And Subscription with name "test_subscription" is added as body
+    When Perform request on endpoint "/subscriptions"
+    Then Request should get response code 200
+    Given A GET request is prepared
+    When Perform request on endpoint "/subscriptions/test_subscription"
+    Then Request should get response code 200
+    Given A DELETE request is prepared
+    When Perform request on endpoint "/subscriptions/test_subscription"
+    Then Request should get response code 200
+    Given A GET request is prepared
+    When Perform request on endpoint "/subscriptions/test_subscription"
+    Then Request should get response code 404
+
+  @Test_Post_Put_Get_Delete_and_Get_(not_found)_subscription
+  Scenario Outline: Perform <type> request on <endpoint> and expect response code <responsecode>
+    Given A <type> request is prepared
+    Given "<add_sub>" add subscription with name "test_subscription" to the request body
+    When Perform request on endpoint "<endpoint>"
+    Then Request should get response code <responsecode>
+
+  Examples: 
+    | responsecode | type   | add_sub | endpoint                         |
+    | 200          | POST   | do      | /subscriptions                   |
+    | 200          | PUT    | do      | /subscriptions                   |
+    | 200          | GET    | do not  | /subscriptions/test_subscription |
+    | 200          | DELETE | do not  | /subscriptions/test_subscription |
+    | 404          | GET    | do not  | /subscriptions/test_subscription |
+

```

Comment:
Note that the Scenario "Test POST PUT GET DELETE on /subscriptions endpoint" does the same test as the Scenario Outline "Perform <type> request on <endpoint> and expect response code <responsecode>" discussion what to keep should be done.

Reply:
I think both are fine. Pick one that suits you.